apps/astarte_housekeeping_api/lib/astarte_housekeeping_api_web/controllers/realm_controller.ex

```diff
@@ -78,10 +78,12 @@ defmodule Astarte.Housekeeping.APIWeb.RealmController do
 
   defp normalize_update_attrs(update_attrs) when is_map(update_attrs) do
     update_attrs
-    |> Map.replace_lazy(:device_registration_limit, &normalize_device_registration_limit/1)
-    |> Map.replace_lazy("device_registration_limit", &normalize_device_registration_limit/1)
+    |> Map.replace_lazy(:device_registration_limit, &normalize_integer_or_nil/1)
+    |> Map.replace_lazy("device_registration_limit", &normalize_integer_or_nil/1)
+    |> Map.replace_lazy(:datastream_maximum_storage_retention, &normalize_integer_or_nil/1)
+    |> Map.replace_lazy("datastream_maximum_storage_retention", &normalize_integer_or_nil/1)
   end
 
-  defp normalize_device_registration_limit(value) when is_nil(value), do: :remove_limit
-  defp normalize_device_registration_limit(value), do: value
+  defp normalize_integer_or_nil(value) when is_nil(value), do: :none
```

Comment:
I'd use `:remove` or `:unset` since this is the actual semantics (and renaming the `normalize` function accordingly). `:none` is unclear since it's too similar to `nil`.

Reply:
`:none` is needed because the cast in `NonNegativeIntegerOrNoneType`. Since both `device_registration_limit` and `datastream_maximum_storage_retention` use it (and the semantics slightly differs), I would stay with `:none`

Reply:
Isn't the semantics "remove the limit if this is `nil`" for both of them? Or am I missing something? The previous iteration was already using `:remove` as a verb

Reply:
Datastream max storage retention uses both 0 and `nil` as valid values for removing the limit (as ScyllaDB uses TTL=0 to mean no TTL), while 0 is a valid value to set the device registration limit to.
A more explicit option would be to make 0 an invalid value for datastream max storage retention, leading to two Ecto custom types (one for { n | n > 0} U {:none} and one for { n | n >= 0} U {:none}), but I think that since this is actually an implementation detail due to the database we're using under the hood, it is better to hide the conversion from the user and perform it here https://github.com/astarte-platform/astarte/pull/912/files#diff-46c1424fa6f1d61351cd534f219215a0681236b66406f9be76e22bbc2b0a22acR287-R298

Reply:
Gone with `:unset` after IRL discussion